lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/LocalRepositorySearcher.java

```diff
@@ -77,36 +72,54 @@ public Set<String> searchPluginGroupIds() throws IOException {
 		return getLocalPluginArtifacts().stream().map(Artifact::getGroupId).distinct().collect(Collectors.toSet());
 	}
 
-	public Collection<Artifact> getLocalPluginArtifacts()  {
-		return getLocalArtifactsLastVersion().stream().filter(gav -> gav.getArtifactId().contains("-plugin")).collect(Collectors.toSet());
+	public Collection<Artifact> getLocalPluginArtifacts() {
+		return getLocalArtifactsLastVersion().stream().filter(gav -> gav.getArtifactId().contains("-plugin"))
+				.collect(Collectors.toSet());
 	}
 
 	public Collection<Artifact> getLocalArtifactsLastVersion() {
-		CompletableFuture<Collection<Artifact>> loadLocalArtifacts = cache.get(localRepository);
-		if (loadLocalArtifacts == null || loadLocalArtifacts.isCompletedExceptionally()) {
-			loadLocalArtifacts = getOrCreateLocalArtifactsLastVersion();
+		if (loadAllLocalArtifacts == null || loadAllLocalArtifacts.isCompletedExceptionally()) {
+			loadAllLocalArtifacts = getOrCreateLocalArtifactsLastVersion();
 		}
-		if (loadLocalArtifacts.isDone()) {
-			return loadLocalArtifacts.getNow(Collections.emptyList());
+		if (loadAllLocalArtifacts.isDone()) {
+			return loadAllLocalArtifacts.getNow(Collections.emptyList());
 		}
 		// The local artifacts search is not finished, returns an empty list
 		return Collections.emptyList();
 	}
 
 	private synchronized CompletableFuture<Collection<Artifact>> getOrCreateLocalArtifactsLastVersion() {
+		if (!(loadAllLocalArtifacts == null || loadAllLocalArtifacts.isCompletedExceptionally())) {
+			return loadAllLocalArtifacts;
+		}
+		loadAllLocalArtifacts = allOf(cache.values());
+		return loadAllLocalArtifacts;
+	}
+
+	private static <T> CompletableFuture<Collection<T>> allOf(Collection<CompletableFuture<Collection<T>>> futures) {
+		return CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()])) //
+				.thenApply(__ -> futures.stream() //
+						.map(CompletableFuture::join) //
+						.flatMap(list -> list.stream()) //
+						.toList());
+	}
+
+	private void createCreateLocalArtifactsFuture(File localRepository) {

```

Comment:
Can this be jsst `createLocalArtifactsFuture(...)`?

Reply:
fixed